### ISSUE TITLE
3.3 is Completely Ignoring Configured Folders and Resetting and Messing up Bookmarks

### ISSUE THREAD

**Issue by GuardianMajor** [eric-bixby/auto-sort-bookmarks-webext#82]
Upgraded to 3.3 with great hopes only to have it completely reset the configured folders and sort everything that it wasn't supposed to and mess up very carefully organized toolbar and menu.

I keep changing it back to what it should be and the minute the options window is closed, it resets back to the default, this is not cool at all, fixing one problem only to screw up something else, so disappointed.

(what it should be)
![image](https://user-images.githubusercontent.com/5274181/41807671-1094e4ca-7687-11e8-9b20-28c8b0c3ea5d.png)

(what it keeps resetting back to)
![image](https://user-images.githubusercontent.com/5274181/41807686-3cf612d2-7687-11e8-9466-5c5d684a554f.png)

it seems every time you check "auto sort" and then save the configuration it resets this, whether you are enabling it, or disabling it.

(enabled and it reset configured folders)
![image](https://user-images.githubusercontent.com/5274181/41807697-7569f5a2-7687-11e8-9912-c299dd2b997d.png)

This is so dumb.

**Comment by GuardianMajor** [comment]
Steps to reproduce it each and every time.

1. Options
2. Configure folders
3. Close
4. Check autosort
5. Save
6. Close
7. Options
8. Check folders (still there)
9. uncheck autosort
10. Save
11. Options
12. Configure folders (all gone)

Do you see it now?

**Comment by shangles** [comment]
I am back to having the issue of it sorting the bookmark toolbar alphabetically even though I told it to exclude.

**Comment by GuardianMajor** [comment]
I don't know how it would be a timing issue, the save should be instant and on top of that there is plenty of wait time for it to get there, so it shouldn't be an issue. It is happening without fail, so I don't know how you can't reproduce it, perhaps you are using a system that somehow works, but it is not here and it is on a clean profile, so there is nothing else that would interfere. Literally nothing. Try it on a Fx 56, maybe it is something that is not accounting for that, it is also happening on the ESR, so it might be something you are not accounting for in older versions of Firefox.

**Comment by eric-bixby** [comment]
Yes, this is a Firefox version specific problem.  I just ran it on FF ESR 52 and saw the problem.

**Comment by shangles** [comment]
I'm on Firefox 60.0.2 (64-bit) in Windows 10, and after uninstalling and reinstalling the add-on, it's back to alphabetically sorting the bookmarks toolbar menu again.

**Comment by GuardianMajor** [comment]
@eric-bixby Great, glad you spotted it. Please let me know when you have a fix for it so I can reinstall and try it again, thanks.

**Comment by shangles** [comment]
A couple things to add.  I just upgraded FF to v61.  In my household I also have a MacBook that I use on my Firefox Sync account.  I wonder if there was some strangeness going on with it syncing the unwanted, alphabetically sorted, bookmarks in the toolbar from the Mac back to my Windows machine?

Not sure.  I've since logged into both computers, upgraded Firefox, reinstalled ASB v3.3 and configured them identically.  So far, the toolbar has the bookmarks manually sorted where I want them.  Hopefully it lasts.
